docs/DeveloperGuide.md

```diff
@@ -6,24 +6,177 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Parser component
+The Parser is essential to interpreting user commands and executing appropriate actions on the timetable. 
+This includes command parsing; breaking down user inputs into commands and parameters that PlaNus can understand and
+act upon.
 
+Parser interacts with the following classes.
+- 'Course': Creates Courses based on user input
+- 'Timetable': Modifies timetable based on user input
+- 'Storage': Uses Storage to save user data
+- 'UI': Uses Ui to display error messages
+
+Below is the class diagram:
+
+![Parser_ClassDiagram](uml/Parser_ClassDiagram.png)
```

Comment:
Id suggest that the different classes in the class diagram do not have this symbol 
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/156929749/187285de-d0ef-41a4-b6a2-b04772d900ed)
Also, the public are to be used with "+" and private is with "-" 
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/156929749/45c881b1-6db7-464d-a9ae-76142febeedd)